bcf_sr_sort.c

```diff
@@ -24,21 +24,30 @@
 
 #include "bcf_sr_sort.h"
 #include "htslib/khash_str2int.h"
+#include <errno.h>
 
 #define SR_REF   1
 #define SR_SNP   2
 #define SR_INDEL 4
 #define SR_OTHER 8
 #define SR_SCORE(srt,a,b) (srt)->score[((a)<<4)|(b)]
 
+static inline void *die_if_null(void *ptr)
+{
+    if ( ptr ) return ptr;
+    errno = ENOMEM;
+    fprintf(stderr, "[E::%s] %s\n", __func__,strerror(errno));
+    exit(1);
```

Comment:
Please don't copy that.  It's there because `hts_expand()` is basically otherwise unfixable (which also means we really shouldn't be using `hts_expand()`, but that's another issue).  See the comments on #442 to see what our users think of this sort of behaviour.

As @jkbonfield says, good libraries should return error codes, rather than calling exit or dumping core.